cmd/banner.go

```diff
@@ -0,0 +1,108 @@
+package cmd
+
+import (
+	"fmt"
+
+	resty "github.com/go-resty/resty/v2"
+	helper "github.com/home-assistant/cli/client"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+const haBanner = `
+| |  | |                          /\           (_)   | |            | |  
+| |__| | ___  _ __ ___   ___     /  \   ___ ___ _ ___| |_ __ _ _ __ | |_ 
+|  __  |/ _ \| '_ \ _ \\ / _ \   / /\ \ / __/ __| / __| __/ _\ | '_ \| __|
+| |  | | (_) | | | | | |  __/  / ____ \\__ \__ \ \__ \ || (_| | | | | |_ 
+|_|  |_|\___/|_| |_| |_|\___| /_/    \_\___/___/_|___/\__\__,_|_| |_|\__|
+
+Welcome to the Home Assistant command line.
+`
+
+const moreInfo = `
+For more details use 'help' and 'exit' to close.
+If you need access to host system use 'login'.
+`
+
+func pokeAPI(section string, command string) (outdata *(map[string]interface{})) {
+	base := viper.GetString("endpoint")
+
+	url, err := helper.URLHelper(base, section, command)
+	if err != nil {
+		fmt.Printf("Error: %v", err)
+		return
+	}
+
+	request := helper.GetJSONRequest()
+	resp, _ := request.Get(url)
+
+	if !resty.IsJSONType(resp.Header().Get("Content-Type")) {
+		fmt.Println("Error: API did not return a JSON response")
+		return
+	}
+
+	var data *helper.Response
+	if resp.IsSuccess() {
+		data = resp.Result().(*helper.Response)
+	} else {
+		data = resp.Error().(*helper.Response)
+	}
+	if data.Result == "ok" {
+		if len(data.Data) == 0 {
+			fmt.Println("Command completed successfully.")
+		} else {
+			outdata = &(data.Data)
+		}
+	}
+	return
+}
+
+var bannerCmd = &cobra.Command{
+	Use:     "banner",
+	Aliases: []string{"ba"},
+	Short:   "Prints the CLI Home Assistant banner along with some useful information",
+	Example: `
+  ha banner
+	`,
+	Run: func(cmd *cobra.Command, args []string) {
+		log.WithField("args", args).Debug("info")
+
+		fmt.Print(haBanner)
+		fmt.Println()
+
+		fmt.Println("System information")
+
+		// Print network address information
+		netinfo := pokeAPI("network", "info")
+
+		netifaces := (*netinfo)["interfaces"].(map[string]interface{})
+
+		for key, value := range netifaces {
+			netiface := value.(map[string]interface{})
+			title := fmt.Sprintf("IPv4 addresss for %s:", key)
+			fmt.Printf("  %-25s %s\n", title, netiface["ip_address"])
+		}
+
+		// Print Host URL
+		hostinfo := *pokeAPI("host", "info")
+		coreinfo := pokeAPI("core", "info")
+
+		protocol := "http"
+		if (*coreinfo)["ssl"] == "true" {
+			protocol = "https"
+		}
+
+		port, _ := (*coreinfo)["port"].(float64)
+		fmt.Printf("  %-25s %s\n", "Operating System Version:", hostinfo["operating_system"])
+		fmt.Printf("  %-25s %s\n", "Home Assistant Core:", (*coreinfo)["version"])
+		fmt.Printf("  %-25s %s://%s:%d\n", "Internal URL:", protocol, hostinfo["hostname"], int(port))
+
+		fmt.Print(moreInfo)
```

Comment:
We should either remove this part or make it optional using a flag.

Reasoning: This text is very specific towards the CLI Plugin. However, this banner 👍 We can use that for e.g., the SSH & Web Terminal add-ons as well (but that this `moreInfo` doesn't apply).